echoregions/regions2d/regions2d.py

```diff
@@ -656,49 +616,48 @@ def transect_mask(
-                    f"{break_str} or {end_str}"
-                )
+        # Reset Transect Dataframe Index
+        transect_df.reset_index(inplace=True)
 
-        # Check if break_str followed by resume_str.
-        break_transect_rows = transect_df[transect_df["transect_type"] == break_str]
-        break_transect_type_next_list = list(break_transect_rows["transect_type_next"].values)
-        for transect_type_next in break_transect_type_next_list:
-            if transect_type_next != resume_str:
-                raise ValueError(
-                    f"Transect break string is followed by invalid value "
-                    f"{transect_type_next}. Must be followed by {resume_str}."
-                )
+        # Create an empty list to collect error messages.
+        warning_messages = []
 
-        # Check if resume_str followed by break_str/end_str.
-        resume_transect_rows = transect_df[transect_df["transect_type"] == resume_str]
-        resume_transect_type_next_list = list(resume_transect_rows["transect_type_next"].values)
-        for transect_type_next in resume_transect_type_next_list:
-            if transect_type_next not in [break_str, end_str]:
-                raise ValueError(
-                    f"Transect resume string is followed by invalid value "
-                    f"{transect_type_next}. Must be followed by either "
-                    f"{break_str} or {end_str}."
-                )
+        # Set transect_type_next values to be empty strings if they are NAs.
+        transect_df["transect_type_next"] = transect_df.apply(
+            lambda x: "" if isna(x["transect_type_next"]) else x["transect_type_next"],
+            axis=1,
+        )
 
-        # Check if end_str followed by start_str or if NA.
-        end_transect_rows = transect_df[transect_df["transect_type"] == end_str]
-        end_transect_type_next_list = list(end_transect_rows["transect_type_next"].values)
-        for transect_type_next in end_transect_type_next_list:
-            # If this value is not NA, check if it is start_str.
-            if not isna(transect_type_next):
-                if transect_type_next != start_str:
-                    raise ValueError(
-                        f"Transect end string is followed by invalid value "
-                        f"{transect_type_next}. Must be followed by {start_str}."
-                    )
+        # Ensure correct sequence of transect types occur. If they do not, append to warning_messages.
+        for index, row in transect_df.iterrows():
+            transect_type = row["transect_type"]
+            transect_type_next = row["transect_type_next"]
+            type_next_warning_message = (
+                f"Error in row {index + 2}: Transect string {transect_type} is followed by "
+                f"invalid value {transect_type_next}. Must be followed by "
+                f"{transect_next_allowable[transect_type]}"
+            )
```

Comment:
It would make more sense to move this section to below the `if` check below, since it is not otherwise used.

Reply:
Also, github conveniently points out that this check is not covered by tests at the moment.

Reply:
Made the first change and I'll also add a test that covers this.